Core/src/org/sleuthkit/autopsy/modules/interestingitems/InterestingItemDefsPanel.java

```diff
@@ -306,7 +306,7 @@ private void doFilesSetRuleDialog(FilesSet.Rule selectedRule) {
                 rules.remove(selectedRule.getName());
             }
             FilesSet.Rule newRule = new FilesSet.Rule(panel.getRuleName(), panel.getFileNameFilter(), panel.getMetaTypeFilter(), panel.getPathFilter());
-            rules.put(newRule.getName(), newRule);
+            rules.put(newRule.toString(), newRule);
```

Comment:
HashMap rules<String, Rule> needs a unique key. newRule.toString is a unique string and an ideal key.